### PR TITLE
server: change ws ping API 

### LINKED ISSUE #1195: [server]: add configuration to disable periodic `Websocket Ping`
https://github.com/paritytech/jsonrpsee/pull/1192#discussion_r1307020404

When we have `PingConfig`, it would be nice to add `PingConfig::Disable` for now it's possible to workaround
but setting the interval to `Duration::MAX` for that to disabled.

### PR DESCRIPTION
**Pull request by niklasad1**
Close #1195

### CODE REVIEW

File: server/src/server.rs [hunk 1]
```diff
@@ -286,28 +286,31 @@ impl ConnectionState {
 pub enum PingConfig {
 	/// The server pings the connected clients continuously at the configured interval but
 	/// doesn't disconnect them if no pongs are received from the client.
-	WithoutInactivityCheck(Duration),
+	OnlyPing(Duration),
```
Comment (niklasad1) [inline server/src/server.rs:289]: I didn't like this verbose names, any suggestions on this?

I still don't like the naming

Comment (tadeohepperle) [reply server/src/server.rs:289]: Hmm, maybe we can join the `OnlyPing` and `Ping` variants into one:
```rs
enum PingConfig {
    Disabled,
    Ping {
        ping_interval: Duration,
        disconnect_after: Option<Duration>
    },
}
```

**Review by tadeohepperle — APPROVED** [review]

**Review by lexnv — APPROVED** [review]
Nice one! 👍

File: server/src/transport/ws.rs [hunk 2]
```diff
@@ -174,15 +176,16 @@ where
...
 ) {
-	// Interval to send out continuously `pings`.
-	let mut ping_interval = tokio::time::interval(ping_interval);
-	// This returns immediately so make sure it doesn't resolve before the ping_interval has been elapsed.
-	ping_interval.tick().await;
-
-	let ping_interval = IntervalStream::new(ping_interval);
+	let ping_interval = match ping_config {
+		None => IntervalStream::pending(),
+		// NOTE: we are emitted a tick here immidiately to sync
```
Comment (niklasad1) [inline server/src/transport/ws.rs:184]:
```suggestion
		// NOTE: we are emitted a tick here immediately to sync
```

**Review by tadeohepperle — APPROVED** [review]
Nice, I like the changes to `PingConfig` :)

File: server/src/server.rs [hunk 3]
```diff
@@ -274,51 +274,66 @@ impl ConnectionState {
...
+/// clients.
 ///
-/// If the server sends out a ping then remote peer must reply with a corresponding pong message.
+/// It's possible to configure how often pings are sent out and how long the server will
+/// wait until a client is determined as "inactive".
 ///
-/// It's possible to just send out pings then don't care about response
-/// or terminate the connection if the ping isn't replied to the configured `max_inactivity` limit.
+/// jsonrpsee doesn't associate the ping/pong frames just that if
+/// pong frame isn't received within `inactive_limit` it's regarded
```
Comment (niklasad1) [inline server/src/server.rs:284]:
```suggestion
/// pong frame isn't received within `inactive_limit` then it's regarded
```

**Review by lexnv — APPROVED** [review]
Nice one!

**Pull request merged**